src/v/raft/voter_priority_tracker.cc

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "raft/voter_priority_tracker.h"
+namespace raft {
+namespace {
+// zero priority doesn't allow node to become a leader
+
+[[maybe_unused]] inline constexpr voter_priority zero_voter_priority
+  = voter_priority{0};
+// 1 is smallest possible priority allowing node to become a leader
+inline constexpr voter_priority min_voter_priority = voter_priority{1};
+inline constexpr voter_priority not_ready_voter_priority = voter_priority{2};
```

Comment:
Wonder why we need separate priority=2 here, can't 1 be used here as well?

Reply:
We need to differentiate between the two cases as when a replica is ready to become voter we reset the priority only if it is different than min_voter_priority. We do not want to reset the priority if a node hosting replica is in maintenance mode.